File: CLIENT/CLIENT.ProcureAccess/src/app/features/criteria/models/create-criterion.dto.ts
import 'reflect-metadata';
import { jsonObject, jsonMember, TypedJSON } from 'typedjson';

@jsonObject
export class CreateCriterionDto {
  @jsonMember
  name: string;

  @jsonMember
  description: string;

  @jsonMember
  criteriaFilterId: number;

  constructor(
    pName: string, 
    pDescription: string,
    pCriteriaFilter: number) {
    this.name = pName;
    this.description = pDescription;
    this.criteriaFilterId = pCriteriaFilter;
  }
}

File: DAL/MODELS.ProcureAccess/Entities/Dto/CreateCriterionDto.cs
namespace MODELS.ProcureAccess.Entities.Dto;

public class CreateCriterionDto
{
    [Required]
    [StringLength(200)]
    public string Name { get; set; }

    [Required]
    [StringLength(6000)]
    public string Description { get; set; }

    public int CriteriaFilterId { get; set; }
}
